lib/nmatrix/lapack.rb

```diff
@@ -1,8 +1,20 @@
 module NumRuby::Linalg
-  def self.inv(obj)
-    if obj.is_a?(NMatrix)
-      return obj.invert
+  def self.inv(matrix)
+    if not matrix.is_a?(NMatrix)
+      raise("Invalid matrix. Not of type NMatrix.")
+    end
+    if matrix.dim != 2
+      raise("Invalid shape of matrix. Should be 2.")
+    end
+    if matrix.shape[0] != matrix.shape[1]
+      raise("Invalid shape. Expected square matrix.")
     end
+    m, n = matrix.shape
+
+    lu, ipiv = NumRuby::Lapack.getrf(matrix)
+    inv_a = NumRuby::Lapack.getri(lu, ipiv)
+
+    return inv_a
```

Comment:
As this is the last line in this function, it isn't necessary to put `return`. You could put `inv_a` and this variable is automatically returned.

Reply:
I believe including `return` makes it more readable.